### ISSUE TITLE
Resnet_flax Downsample2D implementation is not backward compatible with the pre-trained Stable Diffusion v1.4 model

### ISSUE THREAD

**Issue by xcyan** [huggingface/diffusers#5137]
### Describe the bug

TLDR: Resnet_flax Downsample2D implementation is not backward compatible with the pre-trained Stable Diffusion v1.4 model.

I downloaded the Stable diffusion v1.4 model (under the flax branch) and ran model inference with the current implementation. I ended up with very poor generation results.

I noticed that the class [FlaxDownsample2D](https://github.com/huggingface/diffusers/blame/main/src/diffusers/models/resnet_flax.py#L43-L60) used in the UNet is using Conv2D with padding.
```
def setup(self):
        self.conv = nn.Conv(
            self.out_channels,
            kernel_size=(3, 3),
            strides=(2, 2),
            padding=((1, 1), (1, 1)),  # padding="VALID",
            dtype=self.dtype,
        )

    def __call__(self, hidden_states):
        # pad = ((0, 0), (0, 1), (0, 1), (0, 0))  # pad height and width dim
        # hidden_states = jnp.pad(hidden_states, pad_width=pad)
        hidden_states = self.conv(hidden_states)
        return hidden_states
```

However, this is different from the original pytorch implementation of the [Downsample2D layer](https://github.com/CompVis/stable-diffusion/blob/main/ldm/modules/diffusionmodules/model.py#L60-L79).

I tried uncomment the lines and was able to obtain very decent generation results.
```
def setup(self):
        self.conv = nn.Conv(
            self.out_channels,
            kernel_size=(3, 3),
            strides=(2, 2),
            padding="VALID",
            dtype=self.dtype,
        )

    def __call__(self, hidden_states):
        pad = ((0, 0), (0, 1), (0, 1), (0, 0))  # pad height and width dim
        hidden_states = jnp.pad(hidden_states, pad_width=pad)
        hidden_states = self.conv(hidden_states)
        return hidden_states
```

Please fix or at least leave a comment in the code.

### Reproduction

Given the prompt as input: A photograph of an astronaut riding a horse on Mars, high resolution, high definition.

Before the fix:
![stable_diffusion_before](https://github.com/huggingface/diffusers/assets/8760875/fcfbf020-6650-4365-9b2e-cfc5c76c4d53)

After the fix:
![stable_diffusion_after](https://github.com/huggingface/diffusers/assets/8760875/7647921a-41fa-4e09-9b5d-b1b847470505)

### System Info

N / A

### Who can help?

@patrickvonplaten

**Comment by patrickvonplaten** [comment]
Hey @xcyan,

Could you maybe open a PR so that we can directly look into the required code changes? :-) Thanks!

**Comment by patrickvonplaten** [comment]
Hey @xcyan,

Can you add a reproducible code snippet so that we can take a look
